static/js/src/publisher-admin/utils/__tests__/generateReleaseChannelRows.test.tsx

```diff
@@ -0,0 +1,168 @@
+import "@testing-library/jest-dom";
+import { render, screen } from "@testing-library/react";
+import {
+  generateReleaseChannelRows,
+  ResourcesCell,
+  BasesCell,
+} from "../generateReleaseChannelRows";
+import { MainTable } from "@canonical/react-components";
+
+describe("generateReleaseChannelRows", () => {
+  const releaseChannel: ReleaseChannel = {
+    track: "latest",
+    risk: "stable",
+    releases: [
+      {
+        revision: {
+          version: "1.0",
+          bases: [
+            { name: "ubuntu", channel: "18.04", architecture: "amd64" },
+            { name: "centos", channel: "7", architecture: "x86_64" },
+          ],
+          "created-at": "2022-01-01",
+          "sha3-384": "123456",
+          errors: null,
+          revision: 1,
+          size: 12345,
+          status: "published",
+        },
+        resources: [
+          { name: "resource1", type: "oci", revision: 2 },
+          { name: "resource2", type: "file", revision: null },
+        ],
+      },
+      {
+        revision: {
+          version: "2.0",
+          bases: [
+            { name: "ubuntu", channel: "20.04", architecture: "amd64" },
+            { name: "centos", channel: "8", architecture: "x86_64" },
+          ],
+          "created-at": "2022-01-01",
+          "sha3-384": "123456",
+          errors: null,
+          revision: 1,
+          size: 12345,
+          status: "published",
+        },
+        resources: [],
+      },
+    ],
+  };
+
+  test("renders release channel rows correctly", () => {
+    const rows = generateReleaseChannelRows(
```

Comment:
You could add a `beforeEach` block to `generateReleaseChannelRows` and render the `MainTable` to avoid repeating it at the start of each test

Reply:
Sure, makes sense :+1: